pkg/ruler/ruler.proto

```diff
@@ -19,7 +19,12 @@ service Ruler {
   rpc Rules(RulesRequest) returns (RulesResponse) {};
 }
 
-message RulesRequest {}
+message RulesRequest {
+  repeated string ruleNames = 1;
+  repeated string ruleGroupNames = 2;
+  repeated string files = 3;
+  string type = 4;
+}
```

Comment:
Will this cause compatibility issue? During rollout, protocol changes and new/old ruler cannot communicate with each other?

Reply:
If the clients do not send any filters, then the results are consistent during deployment. However, if the clients start sending filters during deployment, the results are inconsistent until the deployment is complete. This is because old ruler does not filter and the new one will